scvi/model/base/_base_model.py

```diff
@@ -224,6 +224,29 @@ def _register_manager_for_instance(self, adata_manager: AnnDataManager):
         instance_manager_store = self._per_instance_manager_store[self.id]
         instance_manager_store[adata_id] = adata_manager
 
+    def deregister_manager(self, adata_manager: Optional[AnnDataManager] = None):
+        """Deregisters an :class:`~scvi.data.AnnDataManager` instance with this model instance.
+
+        If `adata_manager` is `None`, deregisters all :class:`~scvi.data.AnnDataManager` instances
+        in both the class and instance-specific manager stores, except for the current
+        :class:`~scvi.data.AnnDataManager` instance associated with this model instance.
+        """
+        cls_manager_store = self._setup_adata_manager_store
+        instance_manager_store = self._per_instance_manager_store[self.id]
+
+        if adata_manager is None:
+            manager_ids_to_clear = list(cls_manager_store.keys())
+        else:
+            manager_ids_to_clear = [adata_manager.adata_uuid]
+
+        for manager_id in manager_ids_to_clear:
+            if adata_manager is None and manager_id == self.adata_manager.adata_uuid:
+                # don't clear the current manager by default
+                continue
+            if manager_id in instance_manager_store:
+                del instance_manager_store[manager_id]
+            del cls_manager_store[manager_id]
+
```

Comment:
this doesn't capture the case where manager is in the instance but not the cls manager store. 

This arises commonly if a new anndata is passed to a `get...` method.

Therefore we should independently clear the cls and instance stores but always skip over the current.

Reply:
The current changes add calls to `register_manager` wherever the instance manager store is updated, such as in `get_anndata_manager`. This ensures that both manager stores are in sync and that the instance is always a subset of the class one. I guess my question now would be whether this is necessary or something we want to do at all.

Reply:
hmmm I'm not sure if we should add this change here. We can discuss with @justjhong later but I think just for this PR we should deregister without changing how registration works. Wdyt?

Reply:
Sounds good, I'll revert the changes then

Reply:
this would be easier to read if there were two separate for loops, one for the cls manager store and one for the instance manager store